### ISSUE TITLE
Calling setData does not seem to update all the points correctly

### ISSUE THREAD

**Issue by Umaaz** [highslide-software/highcharts.com#3978]
I have just started using Highcharts, but unfortunatly i have come accross this issue. 

Essentially if i create a graph with a data set of [[0,0]], and then at a later point update the data using series.setData. The graph seems to start at 0,0 (1970, as i am using dates) and spans accross to the new data time.

I have created this fiddle that reproduces the problem. http://jsfiddle.net/bdonnell/0bsusr77/5/

If you load the fiddle, the graph will appear with a single point at 0, 1970. Then clicking the update button will set the data. However there are now 3 points in the graph. starting at 0, 1970.

Strangly if you click update again, the zero point goes away, and the graph appears as i would have expected it to.

As a work around i have removed the data:[[0,0]] from the defaut series data, and this resolves the issue. However i am a little concerend that the data is not updating correctly in other cases. Hopefully you can let me know why the graphs are acting in this way.

**Comment by highslide-software** [comment]
Seems like the X axis doesn't redraw at all. A good workaround (and best practice) is _not_ to redraw for every call to `setData`, but only redraw at the end: http://jsfiddle.net/highcharts/0bsusr77/6/

**Comment by Umaaz** [comment]
Yeah i agree that the best practise would be to only redraw the once. The reason i was redrawing on all calls is due to another issue we had, where the navigator would fail with the following error: TypeError: oldData[i].update is not a function

This is a snippet of the code that caused the error, and seems to be something todo with the length of the new datapoints being the same as the old data points.

https://gist.github.com/Umaaz/1f5d9f0c172f5aa9d25a#file-gistfile1-js

For what ever reason redrawing on each step works around this problem. Unfortunatly i cannot seem to reporoduce this in a fiddle. I will be back in this area next week and will let you know if i come up with anything.

**Comment by gildorwang** [comment]
I hit this issue too. And redrawing on each step really worked around it. Thanks @Umaaz !

**Comment by highslide-software** [comment]
@Umaaz Then you should probably run `setData` with the `updatePoints` flag set to true: http://api.highcharts.com/highcharts#Series.setData.

**Comment by Umaaz** [comment]
I did try using the updatePoints flag. The problem is that the error is coming out of the `navigatorSeries.setData(baseSeries.options.data, false);`, which is called within the highcharts code, and not by me. Here is the call stack from FireFox. 

![selection_029](https://cloud.githubusercontent.com/assets/707162/6705978/f7fdb51a-cd59-11e4-8b9e-45cab41eed32.png)

I have tried different combinations of the flags to Series.setData, but it seems unless you set redraw to true, this error ocurs. Atleast for me. Hope this helps track down the issue.

**Comment by Umaaz** [comment]
@gildorwang do you mean you encountered oldData[i].update is not a function? Are you able to reproduce it in a fiddle?

**Comment by HamsterCoder** [comment]
I think I have a similar problem.

It takes two steps to reproduce:
1) Remove point from series using removePoint with animation === false
2) Update the rest of the points with setData with animation === false

Call chart redraw.

Now the problem - that one of the points appears twice in the updated series.

Please see my simple fiddle http://jsfiddle.net/homyasusina/3g65ea5g/1/

P.S I go through the trouble of both removing point and using setData to accomplish animation even if the number of points has changed. And it kind of seems to work except for this bug.
